File: back/src/main/java/fr/sncf/comere/users/rest/responses/UserResponseMapper.java
package fr.sncf.comere.users.rest.responses;

import org.springframework.stereotype.Component;

import fr.sncf.comere.users.models.User;

/**
 * Classe injectable permettant de mapper un {@link User} vers un objet {@link UserResponse}.
 */
@Component
public class UserResponseMapper {

    public UserResponse map(User user){
        return UserResponse.builder()
            .id(user.getId())
            .email(user.getEmail())
            .lastName(user.getLastName())
            .firstName(user.getFirstName())
            .role(user.getRole())
            .dateOfBirth(user.getDateOfBirth())
            .build();
    }
    
}

File: front/src/app/users/services/user-response.mapper.ts
import { Injectable } from "@angular/core"
import { User } from "../models/User"
import { UserRole } from "../models/UserRole"
import { UserResponse } from "./user-response"

@Injectable({ providedIn: 'root' })
export class UserResponseMapper {

    map(userResponse: UserResponse): User {
        return new User({
            ...userResponse,
            dateOfBirth: new Date(userResponse.dateOfBirth),
            role: userResponse.role as UserRole
        })
    }
}